pandevice/base.py

```diff
@@ -1900,6 +1900,174 @@ def element_str(self):
+                    # Already a linked reference, skip
+                    continue
+                elif isinstance(orig_value, basestring):
+                    # Handle value as a string
+                    try:
+                        ref_obj = self.find_referenced(orig_value, path.param, include_predefined)
+                    except err.PanObjectNotFound as e:
+                        if exceptions:
+                            raise e
+                        else:
+                            continue
+                    setattr(self, path.param, ref_obj)
+                else:
+                    # Handle value as a list
+                    for idx, value in enumerate(orig_value):
+                        if isinstance(value, PanObject):
+                            # Already a linked reference, skip
+                            continue
+                        try:
+                            ref_obj = self.find_referenced(value, path.param, include_predefined)
+                        except err.PanObjectNotFound as e:
+                            if exceptions:
+                                raise e
+                            else:
+                                continue
+                        getattr(self, path.param)[idx] = ref_obj
+
+    def find_referenced(self, name, param, include_predefined=False):
+        """Find the object referenced by a param
+
+        Args:
+            name (str): The name of the object referenced
+            param (str): The referencing parameter
+            include_predefined (bool): If object not found in config tree, check for
+                a predefined object on the device
+
+        """
+        paths, stubs, settings = self._build_element_info()
+
+        # Find the VarPath to use
+        for var_path in paths:
+            if var_path.param == param:
+                break
+        else:
+            msg = 'Variable {0} is not present in this version'
+            raise ValueError(msg.format(param))
+
+        if not var_path.references:
+            return
+
+        if name is None:
+            return
+
+        ref_obj = self.nearest_pandevice().find(name, var_path.references, recursive=True)
+        if ref_obj:
+            return ref_obj
+        # Couldn't find the referenced object in this pandevice so check the next deepest
+        # XXX: This could be more efficient since the children of the firewall are scanned twice
+        if self.nearest_pandevice().parent:
+            ref_obj = self.panorama().find(name, var_path.references, recursive=True)
+            if ref_obj:
+                return ref_obj
+        # Couldn't find the referenced object anywhere in the config tree
+        # so check in the predefined objects
+        if include_predefined:
+            for reftype in var_path.references:
+                if reftype in self.nearest_pandevice().predefined.OBJECT_TYPES:
+                    self.nearest_pandevice().predefined.object(name, reftype)
```

Comment:
Is this supposed to be `return self.nearest_pandevice().predefined.object(name, reftype)`?